text/0000-lwc-signals.md

```diff
@@ -0,0 +1,373 @@
+* __Backward Compatibility__: Ensuring backward compatibility during the transition period may necessitate maintaining dual systems, complicating the framework's implementation and usage.
+* __Verbose Code__: Some scenarios, like converting `@wire` methods to store subscriptions, result in more verbose code, which might detract from the developer experience.
+
+### Tooling and Documentation Updates
+
+* __Development Tooling__: All tooling, including IDEs, linters, and debuggers, will need updates to accommodate and optimize for Signals and stores.
+* __Educational Materials__: Comprehensive updates to documentation, tutorials, and example code will be essential to support developers in adopting the new model.
+
+### Potential for Misuse
+
+* __Overuse of Signals__: There's a risk that developers might overuse signals for state management, leading to unnecessarily reactive code, which can be less performant and harder to maintain. This is specially relevant because of the runtime detection of `.value` to maintain backward compatibility with existing components. Any misused on the signals can affect the runtime performance of the component and therefore the overall performance of the application.
+* __Subscription Management__: Incorrect handling of subscriptions could introduce memory leaks or lead to unexpected behaviors if not managed correctly. This is specially important because the store implementation doesn't have access anymore to the component's lifecycle hooks, instead it would just stop receiving updates of the stores piped thru `this.$api`, meaning that we are at the mercy of the garbage collector.
+
+### Framework Compatibility Concerns
+
+* __Integration with Existing Libraries__: While the goal is to enable the use of any store that follows the Signal protocol, real-world usage may reveal compatibility issues, necessitating additional wrapper or adapter code. As of today, there are a number of popular Signal-like libraries out there, and we should make sure that we can integrate with them.
+* __Interop with Non-Signal Components__: Components designed without signals in mind may require additional logic to work within the new reactivity model, potentially leading to a fragmented ecosystem during the transition period.
+
+### Performance Implications
+
+* __Granularity of Reactivity__: The fine-grained nature of Signals may introduce performance concerns if developers do not carefully consider the reactivity scope and frequency of updates.
+* __Proxy Overhead Elimination__: While removing the proxy overhead of `@track` can lead to performance gains, it's essential to validate these improvements across a wide range of use cases to ensure that they hold true in practice.
+
+## Alternatives
+
+### 1. Continued Use of `@track` and `@wire`
```

Comment:
@mburr-salesforce @caridy It would be an interesting option to have wires that pass signals instead of deeply frozen objects. That would open things up for a lot of simplifications when for instance passing wired values into e.g. forms without having to create deep clones. But it would of course also open things up for all the mentioned misuse. Still something to assess further in terms of perf pros and cons maybe. Definitely would have an impact on perf, but also on development models as it would allow e.g. mutations without following established approaches like data down and events up.

Reply:
That would be a choice to make for the owner of the adapter, not LWC.